### DISCUSSION TITLE
Let's talk about Code for IBM i Pro

### DISCUSSION THREAD

**Discussion by worksofliam** [codefori/vscode-ibmi#145] [General]
Hello friends, developers, and contributors.

I wanted to write this post to share my thoughts with you on Code for IBM i and Code for IBM i Pro.

Just today, it was announced that ILEditor 2 would stop being maintained in favour of Code for IBM i. This is a wise move for a lot of reasons. I've been wanting to extend VSCode to work with IBM i stuff for years, and now it's finally happening and it's like a dream come true.

This is where Code for IBM i Pro comes into play. Of course, the business that owns Code for IBM i and ILEditor 2 still would like to make some money. The issue with that is that I just love open-source so much. If it were easy to live and write code and make it all open, I'd be very happy.. that's obviously not the case.

Code for IBM i Pro is merely just a way to support development. I will build some extensions that uses Code for IBM i and offer it to Pro users before, eventually (time frames not set in stone), they are also made open-source.

This really allows me to continue supporting the open-source version and keep working on brand new awesome features at the same time.

I want to be very clear, **we will not be profiting off of any of the open-source contributions**. All of the Pro functionality is stuff that's coming right out of ILEditor 2. I **really** want to be as transparent as I can because I love the community a ton. If anyone has any questions or concerns, please share them here with me - in the open. The Pro version also allows me to make some extra pennies off of support that we will offer with it - outside of the issues.

Contributions from others will remain open-source and will be benefited by all. Some of the Pro features coming from ILEditor 2 include:

* Git client ([which is already open-source](https://github.com/halcyon-tech/git-client-ibmi))
* Source date implementation. ILEditor 2 has had this since the beginning. It took a lot of time to get right. We're going to start it as a Pro feature and eventually, we will publish it here on GitHub.
* Code Coverage. Another feature included in ILEditor 2. I wrote this one a while back. It's awesome and I've been wanting to make it open-source forever.. I just need to implement it first.
* Some other nifty things!

If you want to implement a feature that is in the Pro version and then want to PR it, that would be awesome. As the maintainer, I love seeing these contributions from other developers. I am not against other people working on implementing the pro features (source dates is a good example)

Please let me know your thoughts. Thanks!

**Comment by NielsLiisberg** [comment]
Hey Liam; Will you still have "Code for IBM i" and "Code for IBM i Pro"? How will that work?

**Comment by worksofliam** [comment]
> Hey Liam; Will you still have "Code for IBM i" and "Code for IBM i Pro"? How will that work?

Hey, both will exist. Code for IBM i, the free, open source version will be the base of all the functionality and features.

Code for IBM i is merely a way for people to support the business, in which they will receive some of the ILEditor 2 features before they are released into the open-source version.
